function/result_data.go

```diff
@@ -32,24 +31,22 @@ func (d ResultData) Equal(o ResultData) bool {
 
 // Set saves the result data. The value type must be acceptable for the data
 // type in the result definition.
-func (d *ResultData) Set(ctx context.Context, value any) diag.Diagnostics {
-	var diags diag.Diagnostics
-
+func (d *ResultData) Set(ctx context.Context, value any) *FuncError {
 	reflectValue, reflectDiags := fwreflect.FromValue(ctx, d.value.Type(ctx), value, path.Empty())
 
-	diags.Append(reflectDiags...)
+	funcErr := FuncErrorFromDiags(ctx, reflectDiags)
 
-	if diags.HasError() {
-		return diags
+	if funcErr != nil {
+		return funcErr
 	}
 
 	d.value = reflectValue
 
-	return diags
+	return nil
 }
 
 // Value returns the saved value.
-func (d ResultData) Value() attr.Value {
+func (d *ResultData) Value() attr.Value {
```

Comment:
Nit: Same here.

Reply:
The change was just to make all methods on the `ResultData` struct on a pointer receiver.